packages/omezarr/src/sliceview/loader.ts

```diff
@@ -30,7 +30,7 @@ function visitTilesWithin(idealTilePx: vec2, size: vec2, bounds: box2D, visit: (
         for (let y = Math.floor(boundsInTiles.minCorner[1]); y < Math.ceil(boundsInTiles.maxCorner[1]); y += 1) {
             // all tiles visited are always within both the bounds, and the image itself
             const lo = Vec2.mul([x, y], idealTilePx);
-            const hi = Vec2.add(lo, idealTilePx);
+            const hi = Vec2.min(size, Vec2.add(lo, idealTilePx));
```

Comment:
this is the fix - it limits the visited tiles by the total size of the image. The bug here caused a bunch of math later in the process to be incorrect, because the tiles extended past what was actually possible.